### PR TITLE
Update mainchain CCU transaction

### LINKED ISSUE #7724: Update mainchain CCU command
### Description

- Update schema https://github.com/LiskHQ/lips/blob/main/proposals/lip-0053.md#parameters
- Update [verify()](https://github.com/LiskHQ/lips/blob/main/proposals/lip-0053.md#verification)
- Update [execute()](https://github.com/LiskHQ/lips/blob/main/proposals/lip-0053.md#execution)

### Acceptance Criteria

- Should have all unit tests for verify and execute functions

### Additional Information

- Related PR https://github.com/LiskHQ/lips/pull/148
- Related LIP https://github.com/LiskHQ/lips/blob/main/proposals/lip-0053.md
- Related issue https://github.com/LiskHQ/lisk-sdk/issues/7185

### PR DESCRIPTION
**Pull request by shuse2**
### What was the problem?

This PR resolves #7724 

### How was it solved?

- Update CCU schema
- Add `verifyPartnerChainOutboxRoot`
- Update mainchain CCU `verify` and `execute`

### How was it tested?

- Add tests for `verifyPartnerChainOutboxRoot`
- Update tests for mainchain CCU

### CODE REVIEW

File: framework/src/modules/interoperability/mainchain/commands/cc_update.ts [hunk 1]
```diff
@@ -47,114 +43,48 @@ import {
...
-		const partnerValidatorStore = this.stores.get(ChainValidatorsStore);
-		const partnerValidators = await partnerValidatorStore.get(context, txParams.sendingChainID);
-		// If params contains a non-empty activeValidatorsUpdate and non-empty certificate
-		const validatorsHashValidity = checkValidatorsHashWithCertificate(txParams, partnerValidators);
-		if (validatorsHashValidity.error) {
-			return validatorsHashValidity;
+		const sendingChainAccount = await this.stores
+			.get(ChainAccountStore)
+			.get(context, params.sendingChainID);
+		if (sendingChainAccount.status === ChainStatus.REGISTERED && params.certificate.length === 0) {
```
Comment (ishantiw) [inline framework/src/modules/interoperability/mainchain/commands/cc_update.ts:70]: I think we are missing liveness condition for first CCU with a non-empty inbox update,
```python
if chainAccount(ccu.params.sendingChainID).status == CHAIN_STATUS_REGISTERED:
        verifyLivenessConditionForRegisteredChains(ccu)
```

File: framework/test/unit/modules/interoperability/mainchain/commands/cc_update.spec.ts [hunk 2]
```diff
@@ -261,411 +272,539 @@ describe('CrossChainUpdateCommand', () => {
...
-
-			partnerValidatorsDataVerify = {
-				activeValidators: activeValidatorsVerify,
-				certificateThreshold: BigInt(10),
-			};
-
-			executeContext = {
-				getMethodContext: () => createTransientMethodContext({ stateStore }),
-				getStore: createStoreGetter(stateStore).getStore,
+		it('should reject terminate the chain and add an event when ccm format is invalid', async () => {
```
Comment (ishantiw) [inline framework/test/unit/modules/interoperability/mainchain/commands/cc_update.spec.ts:502]:
```suggestion
		it('should reject and terminate the chain and add an event when ccm format is invalid', async () => {
```

Comment (ishantiw) [reply framework/test/unit/modules/interoperability/mainchain/commands/cc_update.spec.ts:502]: and same for other test case description

File: framework/test/unit/modules/interoperability/mainchain/commands/cc_update.spec.ts [hunk 3]
```diff
@@ -261,411 +272,539 @@ describe('CrossChainUpdateCommand', () => {
...
-				receivingChainID: utils.intToBuffer(2, 4),
-				sendingChainID: defaultSendingChainIDBuffer,
-				status: CCMStatusCode.OK,
-			};
-			const nonMainchainCCMSerialized = codec.encode(ccmSchema, nonMainchainCCM);
-			jest
-				.spyOn(interopUtils, 'computeValidatorsHash')
-				.mockReturnValue(defaultCertificateValues.validatorsHash);
-			jest.spyOn(interopUtils, 'commonCCUExecutelogic').mockReturnValue({} as never);
+		it('should call apply for ccm and add to the inbox where receivign chain is the main chain', async () => {
```
Comment (ishantiw) [inline framework/test/unit/modules/interoperability/mainchain/commands/cc_update.spec.ts:770]:
```suggestion
		it('should call apply for ccm and add to the inbox where receiving chain is the mainchain', async () => {
```

File: framework/test/unit/modules/interoperability/mainchain/commands/cc_update.spec.ts [hunk 4]
```diff
@@ -261,411 +272,539 @@ describe('CrossChainUpdateCommand', () => {
...
-				status: CCMStatusCode.OK,
-			};
-			const mainchainCCMSerialized = codec.encode(ccmSchema, mainchainCCM);
-			jest
-				.spyOn(interopUtils, 'computeValidatorsHash')
-				.mockReturnValue(defaultCertificateValues.validatorsHash);
-			const commonCCUExecutelogicMock = jest
-				.spyOn(interopUtils, 'commonCCUExecutelogic')
-				.mockReturnValue({} as never);
+		it('should call forward for ccm and add to the inbox where receivign chain is not the mainchain', async () => {
```
Comment (ishantiw) [inline framework/test/unit/modules/interoperability/mainchain/commands/cc_update.spec.ts:790]:
```suggestion
		it('should call forward for ccm and add to the inbox where receiving chain is not the mainchain', async () => {
```

File: framework/src/modules/interoperability/base_cross_chain_update_command.ts [hunk 5]
```diff
@@ -17,15 +17,24 @@ import { utils } from '@liskhq/lisk-cryptography';
...
 import { CcmSendSuccessEvent } from './events/ccm_send_success';
 import { ccmSchema, crossChainUpdateTransactionParams } from './schemas';
-import { CrossChainMessageContext } from './types';
+import { CrossChainMessageContext, TokenMethod } from './types';
 
 export abstract class BaseCrossChainUpdateCommand extends BaseInteroperabilityCommand {
 	public schema = crossChainUpdateTransactionParams;
 
+	protected _tokenMethod!: TokenMethod;
+	protected _interopsMethod!: BaseInteroperabilityMethod<never>;
```
[the remark this answers is not in the document]
Comment (shuse2) [reply framework/src/modules/interoperability/base_cross_chain_update_command.ts]: Fixed to use T

**Review by ishantiw — APPROVED** [review]

**Review by mitsuaki-u — APPROVED** [review]

**Pull request merged**